Unity/Champloo/Assets/Scripts/Player/Movement States/InAir.cs

```diff
@@ -57,7 +57,19 @@ public override MovementState UpdateState(ref Vector3 velocity, ref Vector3 exte
 
         if (controller.collisions.Above && velocity.y > 0) velocity.y = 0;
 
-        velocity.y -= player.Gravity*Time.deltaTime;
+
+        //if player releases the jump button,change the momentum
+        //if the player is not holding the jump button and is is moving up
+        if(player.InputPlayer.GetButtonUp("Jump") && velocity.y > 0)
+        {
+            velocity.y = Mathf.Sqrt(velocity.y);
```

Comment:
We might want to consider using something other than Sqrt in the future, unless impact is negligible across network/machines.